VocaDbModel/Service/VideoServices/NicoLogParser.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using HtmlAgilityPack;
+using NLog;
+
+namespace VocaDb.Model.Service.VideoServices {
+	public class NicoLogParser : IVideoServiceParser {
+		public Task<VideoTitleParseResult> GetTitleAsync(string id) => NicoLogHelper.GetVideoTitleParseResultAsync(id);
+	}
+
+	public static class NicoLogHelper {
+		private static readonly Logger log = LogManager.GetCurrentClassLogger();
+
+		private static HtmlDocument GetVideoHtmlPage(string videoId) {
+			var url = string.Format("https://nicolog.jp/watch/{0}", videoId);
+
+			var request = WebRequest.Create(url);
+			request.Timeout = 10000;
+			WebResponse response;
+
+			try {
+				response = request.GetResponse(); // Closed below
+			} catch (WebException x) {
+				log.Warn(x, "Unable to get response for nicolog page");
+				return null;
+			}
+
+			try {
+				using (var stream = response.GetResponseStream()) {
+					var doc = new HtmlDocument();
+					try {
+						doc.Load(stream, Encoding.UTF8);
+					} catch (IOException x) {
+						log.Warn(x, "Unable to load document for nicolog page");
+					}
+
+					return doc;
+				}
+			} finally {
+				response.Close();
+			}
+		}
+
+		public static Task<VideoTitleParseResult> GetVideoTitleParseResultAsync(string id) {
+			HtmlDocument doc = GetVideoHtmlPage(id);
+			VideoTitleParseResult videoTitleParseResult = ParsePage(doc);
+			
+			return Task.FromResult(videoTitleParseResult);
+		}
+
+		public static VideoTitleParseResult ParsePage(HtmlDocument doc) {

```

Comment:
Since the `doc` argument may be null, `HtmlDocument` should be replaced with `HtmlDocument?`.